### PR TITLE
Proposal for WiFiAccessPoint class.

### PR DESCRIPTION
**Pull request by vgolovanov**
## Description
Proposal for WiFiAccessPoint class to be able use ESP32 targer as access point.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (work on Unit Tests, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Windows.Devices.WiFi/WiFiStation.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        PHY802_11lr = 8,
+    };
+
+    //Duplicate of WirelessAPStation class from nanoFramework.System.Net
+    public class WiFiStation
+    {
+        public byte[] MacAddres;
+        public sbyte Rssi;
+        public PhyProtocols PhyModes;
+        //Not sure, but I didn't find any low level code for the IP address of the station info.
```
Comment (josesimoes) [inline Windows.Devices.WiFi/WiFiStation.cs:39]: It doesn't exist. It's in the network interface.

**Comment by josesimoes** [comment]
Considering all that's required with this, I leave 2 suggestions for thought:

1. Maybe this should depend on System.Net because all that's required from there, now duplicated here. Plus, I feel that IP addresses are coming, so....
2. As this is growing it would be wise to have a second NuGet with Wi-Fi + Access Point. That way we keep the current one as slim as possible and don't force the AP to people not needing it.

**Comment by vgolovanov** [comment]
> Considering all that's required with this, I leave 2 suggestions for thought:
> 
> 1. Maybe this should depend on System.Net because all that's required from there, now duplicated here. Plus, I feel that IP addresses are coming, so....
Ok. I will look for the code. With dependency on System.Net i think will be much easier,
> 2. As this is growing it would be wise to have a second NuGet with Wi-Fi + Access Point. That way we keep the current one as slim as possible and don't force the AP to people not needing it.
What about creating a separate class as Windows.Devices.WiFiAccessPoint. Or is that way too much ? :-)

**Comment by josesimoes** [comment]
> What about creating a separate class as Windows.Devices.WiFiAccessPoint. Or is that way too much ? :-)

Really dependes on where this is going... If the AP class brings in a lot of stuff and distinct of what is here, that could make sense. If not, then a separate package with the AP feature should be good enough.

**Comment by josesimoes** [comment]
@AdrianSoundy thoughts on all this?

**Comment by vgolovanov** [comment]
Most usage of WiFiAccessPoint is for the device configuration. Connecting the devices to the wifi network without hardcoded credentials. On the Arduino platform WifiManager is very much in demand.

> > What about creating a separate class as Windows.Devices.WiFiAccessPoint. Or is that way too much ? :-)
> 
> Really dependes on where this is going... If the AP class brings in a lot of stuff and distinct of what is here, that could make sense. If not, then a separate package with the AP feature should be good enough.

**Comment by AdrianSoundy** [comment]
Its a good idea to have a simpler way to start the SoftAp but maybe we should try and get away from the Windows.Devices UWP namespace.  I would rather look at extending the existing System.Net support as in:-

- Changes to allow configuration changes without having to reload, keeping backward compatibility
- Simple call to start SoftAP in the same way as Open method here. This also applies to Wifi station.
- As a separate assembly that extends on the current System.Net / System.NetInformation. Something like System.Net.SoftAP or WiFi

We already have a most of the SoftAP implementation which you are duplicating here, the main problem is the need to reload to start it running.

**Pull request closed without merging**